ui/src/app/applications/components/application-status-panel/application-status-panel.tsx

```diff
@@ -58,60 +58,124 @@ const sectionHeader = (info: SectionInfo, onClick?: () => any) => {
     );
 };
 
-const hasRollingSyncEnabled = (application: models.Application): boolean => {
-    return application.metadata.ownerReferences?.some(ref => ref.kind === 'ApplicationSet') || false;
+const getApplicationSetOwnerRef = (application: models.Application) => {
+    return application.metadata.ownerReferences?.find(ref => ref.kind === 'ApplicationSet');
 };
 
-const ProgressiveSyncStatus = ({application}: {application: models.Application}) => {
-    if (!hasRollingSyncEnabled(application)) {
-        return null;
-    }
+const extractErrorText = (error: any): string => {
+    const errorChildren = error.props?.children || [];
+    return Array.isArray(errorChildren) ? errorChildren.map(child => (typeof child === 'string' ? child : '')).join('') : String(errorChildren);
+};
 
-    const appSetRef = application.metadata.ownerReferences.find(ref => ref.kind === 'ApplicationSet');
+const isPermissionError = (error: any): boolean => {
+    const errorText = extractErrorText(error);
+    return errorText.includes('Failed to load data');
+};
+
+const ProgressiveSyncStatus = ({application}: {application: models.Application}) => {
+    const appSetRef = getApplicationSetOwnerRef(application);
     if (!appSetRef) {
         return null;
     }
 
     return (
         <DataLoader
             input={application}
+            errorRenderer={(error: any) => {
+                // If user doesn't have permission to read ApplicationSet, hide the panel
+                if (isPermissionError(error)) {
+                    return null;
+                }
+                // For other errors, show a minimal error state
+                return (
+                    <div className='application-status-panel__item'>
+                        {sectionHeader({
+                            title: 'PROGRESSIVE SYNC',
+                            helpContent: 'Shows the current status of progressive sync for applications managed by an ApplicationSet with RollingSync strategy.'
+                        })}
+                        <div className='application-status-panel__item-value'>
+                            <i className='fa fa-exclamation-triangle' style={{color: COLORS.sync.unknown}} /> Error
+                        </div>
+                        <div className='application-status-panel__item-name'>Unable to load Progressive Sync status</div>
+                    </div>
+                );
+            }}
             load={async () => {
                 const appSet = await services.applications.getApplicationSet(appSetRef.name, application.metadata.namespace);
```

Comment:
As pointed out in [this](https://github.com/argoproj/argo-cd/issues/24190#issuecomment-3249522964) comment, the current implementations is expecting the applicationSet in the same namespace application - which may not be the case and would cause errors, it would not really be access issue though. 
It's not introduced in this PR, but something to think about?